cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -969,6 +985,57 @@ public ApplicationMeta getLatest(ApplicationReference appRef) {
     });
   }
 
+  @Override
+  public void updateSourceControlMeta(SourceControlMetadataRecord repoSourceControlMeta) {
+    TransactionRunners.run(transactionRunner, context -> {
+      ApplicationReference appRef = new ApplicationReference(repoSourceControlMeta.getNamespace(),
+          repoSourceControlMeta.getName());
+      RepositorySourceControlMetadataStore repoMetadataStore = getRepoSourceControlMetadataStore(
+          context);
+      NamespaceSourceControlMetadataStore namespaceMetadataStore = getNamespaceSourceControlMetadataStore(
+          context);
+      SourceControlMeta namespaceSourceControlMeta = namespaceMetadataStore.get(appRef);
+
+      if (namespaceSourceControlMeta == null || namespaceSourceControlMeta.getFileHash() == null) {
+        repoMetadataStore.write(appRef, false, 0L);
+        return;
+      }
+      Boolean isSynced = namespaceSourceControlMeta.getFileHash()
+          .equals(repoSourceControlMeta.getSpecificationHash());
+      if (isSynced) {
+        namespaceMetadataStore.write(appRef,
+            new SourceControlMeta(namespaceSourceControlMeta.getFileHash(),
+                namespaceSourceControlMeta.getCommitId(),
+                namespaceSourceControlMeta.getLastSyncedAt(), isSynced));
+        repoMetadataStore.write(appRef,
+            isSynced, namespaceSourceControlMeta.getLastSyncedAt().toEpochMilli());
+      } else {
+        repoMetadataStore.write(appRef, isSynced, 0L);
+      }
+    });
+  }
+
+  @Override
+  public int scanRepositorySourceControlMetadata(ScanSourceControlMetadataRequest request,
+      Consumer<SourceControlMetadataRecord> consumer) {
+    return TransactionRunners.run(transactionRunner, context -> {
+      String type = EntityType.APPLICATION.toString();
+      return getRepoSourceControlMetadataStore(context).scan(request, type,
+          consumer);
+    });
+  }
+
+  @Override
+  public int scanNamespaceSourceControlMetadata(ScanSourceControlMetadataRequest request,
```

Comment:
name it scanAppSourceControlMetadata as Default store should contain all app related functionalities

Reply:
Sure, done